src/pages/our-projects.tsx

```diff
@@ -0,0 +1,44 @@
+import OurWork from '@/features/our-work';
+import Footer from '@/features/shared/components/footer';
+import Header from '@/features/shared/components/header';
+import {Container} from '@mui/material';
+import styles from './styles.module.scss';
+import GetInTouchSection from '@/features/home/components/get-in-touch-section';
+import {GetServerSidePropsContext} from 'next';
+import {withIronSessionSsr} from 'iron-session/next';
+import {ironSessionOptions} from '@/lib/utils/iron-session';
+import {UserData} from '@/features/shared/types';
+import OurProjects from '@/features/our-projects';
+
+type Props = {
+  user: UserData;
+};
+
+export default function OurProjectsPage({user}: Props) {
+  return (
+    <Container className={styles.pageContainer}>
+      <Header user={user} />
+      <OurProjects user={user} />
+      <Footer />
+    </Container>
+  );
+}
+
+export const getServerSideProps = withIronSessionSsr(async function (ctx: GetServerSidePropsContext) {
+  const userData = ctx.req.session.user as UserData;
+
+  if (!userData) {
+    return {
+      redirect: {
+        destination: '/login',
+        permanent: false,
+      },
+    };
+  }
+
+  return {
+    props: {
+      user: userData ?? null,
+    },
+  };
+}, ironSessionOptions);
```

Comment:
What the heck is this?

Reply:
It is a redirecting before rendering the page if you are not logged in and you try to go to this page you'll be redirected to the login

Reply:
Cool